components/layout/Header.tsx

```diff
@@ -1,191 +1,212 @@
-import CloseIcon from '@mui/icons-material/Close';
-import Menu from '@mui/icons-material/Menu';
+import { ArrowForwardIos, ChevronRight, Menu } from '@mui/icons-material';
 import {
   AppBar,
   Box,
   Button,
-  Container,
   Drawer,
-  Grid,
   IconButton,
   List,
   ListItem,
   ListItemButton,
   ListItemText,
-  Toolbar,
-  Typography,
 } from '@mui/material';
 import { useTheme } from '@mui/material/styles';
 import useMediaQuery from '@mui/material/useMediaQuery';
 import Link from 'next/link';
 import React, { useState } from 'react';
 
 import navItems from '../../content/navItems.ts';
-import SkipLink from '../helper/SkipLink.tsx';
+import EligibilityButton from '../helper/EligibilityButton.tsx';
 import NavigationLogo from './NavigationLogo.tsx';
 
-export default function Header() {
+interface HeaderProps {
+  isCalc: boolean;
+}
+
+export default function Header({ isCalc }: HeaderProps) {
   const theme = useTheme();
   const matches = useMediaQuery(theme.breakpoints.down('md'));
 
-  const [mobileOpen, setMobileOpen] = useState(false);
+  const [drawerOpen, setDrawerOpen] = useState(false);
+  const [hoveredIndex, setHoveredIndex] = useState<number | null>(null);
 
   const handleDrawerToggle = () => {
-    setMobileOpen(!mobileOpen);
+    setDrawerOpen(!drawerOpen);
+  };
+
+  const handleMouseEnter = (index: number) => {
+    setHoveredIndex(index);
+  };
+
+  const handleMouseLeave = () => {
+    setHoveredIndex(null);
   };
 
   const drawer = (
-    <Box onClick={handleDrawerToggle} sx={{ flexGrow: 1 }}>
-      <Grid item xs={12} sx={{ textAlign: 'right', mr: 0.5 }}>
+    <Box onClick={handleDrawerToggle}>
+      <Box>
         <IconButton
-          color="inherit"
           aria-label="Close navigation"
+          sx={{
+            marginRight: 'auto',
+          }}
         >
-          <CloseIcon fontSize="large" />
+          <ArrowForwardIos
+            fontSize="large"
+            sx={{
+              color: theme.palette.text.light,
+            }}
+          />
         </IconButton>
-      </Grid>
+      </Box>
       <List className="nav-mobile" sx={{ transform: 'translateY(-20px)' }}>
-        {navItems
-          .filter(
-            (item) => item.text !== 'Access Calculator',
-          )
-          .map((item) => (
-            <ListItem key={item.text}>
+        {navItems.map(({ href, text, sublist }) => (
```

Comment:
We need to add a key prop to this iterator.

Reply:
@kathyye15  I can do this